eth/v1/validator_service.proto

```diff
@@ -0,0 +1,218 @@
+option java_package = "org.ethereum.eth.v1";
+option php_namespace = "Ethereum\\Eth\\v1";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+
+import "eth/v1/attestation.proto";
+import "eth/v1/beacon_block.proto";
+import "eth/v1/beacon_state.proto";
+import "eth/v1/validator.proto";
+
+// Beacon chain validator API
+//
+// The beacon chain validator API is a set of endpoints to be used by validators for performing their roles.
+//
+// This service is defined in the upstream eth2.0-apis repository (eth2.0-APIs/apis/validator).
+service BeaconValidator {
+  // GetAttesterDuties requests the beacon node to provide a set of attestation duties, which should be performed
+  // by validators, for a particular epoch. Duties should only need to be checked once per epoch, however
+  // a chain reorganization (of > MIN_SEED_LOOKAHEAD epochs) could occur, resulting in a change of duties.
+  // For full safety, you should monitor chain reorganization events.
+  rpc GetAttesterDuties(AttesterDutiesRequest) returns (AttesterDutiesResponse) {
+    option (google.api.http) = { get: "/eth/v1/validator/duties/attester/{epoch}" };
+  }
+
+  // GetProposerDuties requests beacon node to provide all validators that are scheduled to
+  // propose a block in the given epoch
+  rpc GetProposerDuties(ProposerDutiesRequest) returns (ProposerDutiesResponse) {
+    option (google.api.http) = { get: "/eth/v1/validator/duties/proposer/{epoch}" };
+  }
+
+  // GetBlock requests the beacon node to produce a valid unsigned beacon block,
+  // which can then be signed by a proposer and submitted.
+  rpc GetBlock(ProposerBlockRequest) returns (ProposerBlockResponse) {
+    option (google.api.http) = { get: "/eth/v1/validator/blocks/{slot}" };
+  }
+
+  // GetAttestationData requests that the beacon node provides the attestation data for
+  // the requested committee index and slot based on the nodes current head.
+  rpc GetAttestationData(AttestationDataRequest) returns (AttestationDataResponse) {
+    option (google.api.http) = { get: "/eth/v1/validator/attestation_data" };
+  }
+
+  // GetAggregateAttestation aggregates all attestations matching the given attestation data root and slot,
+  // returning the aggregated result.
+  rpc GetAggregateAttestation(AggregateAttestationRequest) returns (AttestationResponse) {
+    option (google.api.http) = { get: "/eth/v1/validator/aggregate_attestation" };
+  }
+
+  // SubmitAggregateAndProofs verifies given aggregate and proofs and publishes them on appropriate gossipsub topic.
+  //
+  // Response usage:
+  // - 200: Successful response
+  //
+  // - 400: Invalid request syntax.
+  //
+  // - 500: Beacon node internal error.
+  rpc SubmitAggregateAndProofs(AggregateAndProofsSubmit) returns (google.protobuf.Empty) {
+    option (google.api.http) = { post: "/eth/v1/validator/aggregate_attestation" };
```

Comment:
I think you need to add the `body` property.
https://github.com/googleapis/googleapis/blob/8d73f9486fc193a150f6c907dfb9f49431aff3ff/google/api/http.proto#L352

Since this is a simple array from JSON and we want to map it to the data field. I _think_ we can make this happen with `body: "data"`...

Reply:
```suggestion
    option (google.api.http) = { 
      post: "/eth/v1/validator/aggregate_attestation"
      body: "data"
    };
```